File: app/src/main/java/digital/euforia/app/ui/subscription/subs2/Subscription2Config.java
package digital.euforia.app.ui.subscription.subs2;

import com.google.gson.Gson;
import com.google.gson.annotations.SerializedName;
import com.google.gson.reflect.TypeToken;
import digital.euforia.app.R;
import digital.euforia.app.ui.subscription.RC;
import digital.euforia.app.ui.subscription.UserActivity;
import digital.euforia.app.ui.subscription_c.config.ReviewSubscriptionModel;
//import digital.euforia.app.model.service.ReviewSubscriptionModel;
//import digital.euforia.app.ui.UserActivity;
//import digital.euforia.app.util.base.RC;

import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.Reader;
import java.io.Serializable;
import java.io.StringWriter;
import java.io.Writer;
import java.nio.charset.StandardCharsets;
import java.util.ArrayList;

/**
 * Developed by oncreate.com.
 * Copyright © 2019-2023 ONCREATE. All Rights Reserved.
 */

public class Subscription2Config implements Serializable {

    @SerializedName("title")
    public String title;
    @SerializedName("video")
    public String video;
    @SerializedName("music")
    public String music;
    @SerializedName("primaryButtonTitle")
    public String primaryButtonTitle;
    @SerializedName("primaryBottomButtonTitle")
    public String primaryBottomButtonTitle;
    @SerializedName("primaryButtonSubtitle")
    public String primaryButtonSubtitle;
    @SerializedName("primaryProduct")
    public String primaryProduct;
    @SerializedName("footerText")
    public String footerText;
    @SerializedName("plansTitle")
    public String plansTitle;
    @SerializedName("benefitsTitle")
    public String benefitsTitle;
    @SerializedName("extraBenefitsTitle")
    public String extraBenefitsTitle;
    @SerializedName("benefits")
    public ArrayList<String> benefits;
    @SerializedName("plans")
    public ArrayList<PlanConfig> plans;
    @SerializedName("extraBenefits")
    public ArrayList<BenefitConfig> extraBenefits;

    public ArrayList<ReviewSubscriptionModel> reviews;

    public static Subscription2Config newInstance(UserActivity activity) {
        try {
            Subscription2Config config = new Gson().fromJson(RC.getString(activity.getRC(), "premium_2_configuration"), Subscription2Config.class);

            InputStream inputStream = activity.getResources().openRawResource(R.raw.premium_reviews);
            Writer writer = new StringWriter();
            char[] buffer = new char[1024];
            try {
                Reader reader = new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8));
                int n;
                while ((n = reader.read(buffer)) != -1) {
                    writer.write(buffer, 0, n);
                }
            } finally {
                inputStream.close();
            }

            String jsonReviews = writer.toString().trim();

            config.reviews = new Gson().fromJson(jsonReviews, new TypeToken<ArrayList<ReviewSubscriptionModel>>() {
            }.getType());
            return config;
        } catch (Exception e) {
            return new Subscription2Config();
        }
    }
}

File: app/src/main/java/digital/euforia/app/ui/subscription_c/config/Subscription2Config.kt
package digital.euforia.app.ui.subscription_c.config

import com.google.gson.annotations.SerializedName
import java.io.Serializable

/**
 * Developed by oncreate.com.
 * Copyright © 2019-2023 ONCREATE. All Rights Reserved.
 */

data class Subscription2Config(
    @SerializedName("title")
    var title: String = "",
    @SerializedName("video")
    var video: String = "",
    @SerializedName("music")
    var music: String = "",
    @SerializedName("primaryButtonTitle")
    var primaryButtonTitle: String = "",
    @SerializedName("primaryBottomButtonTitle")
    var primaryBottomButtonTitle: String = "",
    @SerializedName("primaryButtonSubtitle")
    var primaryButtonSubtitle: String = "",
    @SerializedName("primaryProduct")
    var primaryProduct: String = "",
    @SerializedName("footerText")
    var footerText: String = "",
    @SerializedName("plansTitle")
    var plansTitle: String = "",
    @SerializedName("benefitsTitle")
    var benefitsTitle: String = "",
    @SerializedName("extraBenefitsTitle")
    var extraBenefitsTitle: String = "",
    @SerializedName("benefits")
    var benefits: ArrayList<String> = ArrayList(),
    @SerializedName("plans")
    var plans: ArrayList<PlanConfig> = ArrayList(),
    @SerializedName("extraBenefits")
    var extraBenefits: ArrayList<BenefitConfig> = ArrayList(),
    var reviews: ArrayList<ReviewSubscriptionModel> = ArrayList()
) : Serializable

data class BenefitConfig(
    @SerializedName("imageUrl")
    var imageUrl: String? = null,
    @SerializedName("text")
    var text: String? = null,
    @SerializedName("title")
    var title: String? = null
) : Serializable

data class PlanConfig(
    @SerializedName("productIdentifier")
    var productIdentifier: String? = null,
    @SerializedName("badge")
    var badge: String? = null,
    @SerializedName("title")
    var title: String? = null,
    @SerializedName("details")
    var details: String? = null,
    @SerializedName("buttonTitle")
    var buttonTitle: String? = null,
    @SerializedName("buttonSubtitle")
    var buttonSubtitle: String? = null
) : Serializable

data class ReviewSubscriptionModel(
    @SerializedName("name")
    var name: String? = null,
    @SerializedName("text")
    var text: String? = null,
    @SerializedName("rating")
    var rating: Int = 5
) : Serializable

